File: library/src/main/java/com/senierr/adapter/internal/UnregisteredException.java
package com.senierr.adapter.internal;

/**
 * 类型未注册异常
 *
 * @author zhouchunjie
 * @date 2018/11/05
 */
public class UnregisteredException extends IllegalArgumentException {

    public UnregisteredException(Class cls) {
        super("Have you registered " + cls.getSimpleName() + ".class?");
    }
}

File: multitype/src/main/java/com/senierr/adapter/internal/UnregisteredException.kt
package com.senierr.adapter.internal

/**
 * 类型未注册异常
 *
 * @author zhouchunjie
 * @date 2018/11/05
 */
class UnregisteredException(cls: Class<*>) : IllegalArgumentException("Have you registered " + cls.simpleName + ".class?")
